### PR TITLE
feat: Add autofocus to readme

### PR DESCRIPTION
**Pull request by KaiPrince**
[US140739](https://rally1.rallydev.com/#/?detail=/userstory/641721547587&fdp=true): UI Core dialog - Add autofocus feature

### CODE REVIEW

File: components/dialog/README.md [hunk 1]
```diff
@@ -256,6 +256,20 @@ document.querySelector('#open').addEventListener('click', () => {
...
+
+### Specifying an `autofocus` Element (Optional)
+
+To specify which element should be focused, add the `autofocus` attribute to that element. An element with the `autofocus` attribute will receive focus if any of the following are true:
+
+* The element has a `tabindex` value of `0` or `-1`
+* The element has a focusable descendant in the plain DOM or its Shadow DOM
+* The element is a naturally focusable element (e.g. button)
+
+Note that the element must be in the dialog's immediate shadow root and not within another component's Shadow DOM.
```
Comment (dbatiste) [inline components/dialog/README.md:271]:
```suggestion
Note that the element must be in the dialog's local DOM scope and not within another component's Shadow DOM.
```

File: components/dialog/README.md [hunk 2]
```diff
@@ -256,6 +256,20 @@ document.querySelector('#open').addEventListener('click', () => {
...
+## Focus Management
+
+When opened, focus will be automatically placed within the dialog. The element to be focused will either be the content element having the optional `autofocus` attribute, or a focusable element identified by the dialog depending on the type of dialog. For `d2l-dialog` and `d2l-dialog-fullscreen`, the first focusable element will be focused. For `d2l-dialog-confirm`, the least destructive action will be focused, which is assumed to be the first non-primary button in the footer.
+
+### Specifying an `autofocus` Element (Optional)
+
+To specify which element should be focused, add the `autofocus` attribute to that element. An element with the `autofocus` attribute will receive focus if any of the following are true:
+
+* The element has a `tabindex` value of `0` or `-1`
+* The element has a focusable descendant in the plain DOM or its Shadow DOM
```
[the remark this answers is not in the document]
Comment (KaiPrince) [reply components/dialog/README.md:268]: > Is this mean to address the scenario where they are applying autofocus to a custom element, which has a focus() method to delegate focus internally?

That scenario is covered under the first bullet (`tabindex`), I believe.

[the remark this answers is not in the document]
Comment (dbatiste) [reply components/dialog/README.md:268]: > This is meant to address the scenario where a custom element puts autofocus on an element which is in their shadow dom.

I don't think we're supporting that scenario.

[the remark this answers is not in the document]
Comment (KaiPrince) [reply components/dialog/README.md:268]: I was looking at https://github.com/BrightspaceUI/core/blob/db2d710becbf612c2757f217de05dd3c64d27464/helpers/focus.js#L113

So actually yes, it is exactly what you said.

[the remark this answers is not in the document]
Comment (KaiPrince) [reply components/dialog/README.md]: > > This is meant to address the scenario where a custom element puts autofocus on an element which is in their shadow dom.
> 
> I don't think we're supporting that scenario.

Yeah we're not, I got confused and was thinking about the note below.

[the remark this answers is not in the document]
Comment (KaiPrince) [reply components/dialog/README.md]: Fixed in https://github.com/BrightspaceUI/core/pull/2621/commits/2b5c95421178d673f06ffc187bc3d61d5d125e66

**Review by dbatiste — APPROVED** [review]

**Review by dlockhart — APPROVED** [review]
Haha I missed the original revisions but whatever you and Dave B did made this look perfect to me!

**Pull request merged**